cmd/ssh.go

```diff
@@ -18,141 +19,47 @@ import (
 	"github.com/unweave/unweave/api/types"
 )
 
-func copySource(execID, rootDir, dstPath string, connectionInfo types.ConnectionInfo, keyPath string) error {
-	name := fmt.Sprintf("uw-context-%s.tar.gz", execID)
-	tmpFile, err := os.CreateTemp(os.TempDir(), name)
-	if err != nil {
-		return err
-	}
-
-	ui.Infof("🧳 Gathering context from %q", rootDir)
-
-	if err := gatherContext(rootDir, tmpFile, "tar"); err != nil {
-		return fmt.Errorf("failed to gather context: %v", err)
-	}
-
-	tmpDstPath := filepath.Join("/tmp", name)
-	scpCommand := exec.Command(
-		"scp",
-		"-r",
-		"-o", "StrictHostKeyChecking=no",
-		"-o", "UserKnownHostsFile=/dev/null",
-		tmpFile.Name(),
-		fmt.Sprintf("%s@%s:%s", connectionInfo.User, connectionInfo.Host, tmpDstPath),
-	)
-	if keyPath != "" {
-		scpCommand.Args = append(scpCommand.Args, "-i", keyPath)
-	}
-
-	stdout := &bytes.Buffer{}
-	stderr := &bytes.Buffer{}
-
-	scpCommand.Stdout = stdout
-	scpCommand.Stderr = stderr
-
-	ui.Infof("🔄 Copying source to %q", dstPath)
-
-	if err := scpCommand.Run(); err != nil {
-		if exitError, ok := err.(*exec.ExitError); ok {
-			// Exited with non-zero exit code
-			if status, ok := exitError.Sys().(syscall.WaitStatus); ok {
-				if status.ExitStatus() == 255 {
-					ui.Infof("The remote host closed the connection.")
-					return fmt.Errorf("failed to copy source: %w", err)
-				}
-			}
-			ui.Infof("Failed to copy source directory to remote host: %s", stderr.String())
-			return err
-		}
-		return fmt.Errorf("scp command failed: %v", err)
-	}
-
-	// Unzip on remote host
-	sshCommand := exec.Command(
-		"ssh",
-		"-o", "StrictHostKeyChecking=no",
-		"-o", "UserKnownHostsFile=/dev/null",
-		"-i", keyPath,
-		fmt.Sprintf("%s@%s", connectionInfo.User, connectionInfo.Host),
-		fmt.Sprintf("tar -xzf %s -C %s && rm -rf %s", tmpDstPath, dstPath, tmpDstPath),
-	)
-
-	sshCommand.Stdout = stdout
-	sshCommand.Stderr = stderr
-
-	if err := sshCommand.Run(); err != nil {
-		if exitError, ok := err.(*exec.ExitError); ok {
-			// Exited with non-zero exit code
-			if status, ok := exitError.Sys().(syscall.WaitStatus); ok {
-				if status.ExitStatus() == 255 {
-					ui.Infof("The remote host closed the connection.")
-					return fmt.Errorf("failed to copy source: %w", err)
-				}
-			}
-			ui.Infof("Failed to extract source directory on remote host: %s", stderr.String())
-			return err
-		}
-		return fmt.Errorf("failed to unzip on remote host: %v", err)
-	}
-
-	ui.Infof("✅  Successfully copied source directory to remote host")
-
-	return nil
+// SSH handles the Cobra command for SSH
+func SSH(cmd *cobra.Command, args []string) error {
+	return sessionConnect(cmd, false, args)
 }
 
-func SSH(cmd *cobra.Command, args []string) error {
+// sessionConnect handles the flow to spawn a new SSH connection to an exec.
+func sessionConnect(cmd *cobra.Command, withOpenVSCode bool, args []string) error {
```

Comment:
This is great to split this but I I think the vscode functionality should not be in here. This should probably return a `ready` chan that the caller can wait on and then call `vscode-remote`. The vscode specific functionality should live in the cobra Code handler

Reply:
Resolved by returning the execCh